### PR TITLE
Trace: construct a CCT

### PR DESCRIPTION
**Pull request by adityaranjan**
1. implementation of cct
2. verified with external tools like vampir for otf2 traces

### CODE REVIEW

**Review by movsesyanae — APPROVED** [review]

**Review by jordanmarry — APPROVED** [review]

**Comment by jordanmarry** [comment]
Rename columns. Location -> Process ID and so on

**Review by jordanmarry — APPROVED** [review]

**Comment by bhatele** [comment]
We need to test this with nsight data

File: pipit/trace.py [hunk 1]
```diff
@@ -29,3 +36,216 @@ def from_hpctoolkit(dirname):
...
+        Generic function to iterate through the trace events and create a CCT.
+        Uses pipit's graph data structure for this. Populates the trace's CCT
+        field and creates a new column in the trace's Events DataFrame that stores
+        a reference to each row's corresponding node in the CCT.
+
+        Thoughts/Concerns:
+        Currently, the DataFrame index of the entry row is being stored
+        in the node's calling context ids. This doesn't really have much of a
+        purpose right now. What should we be storing as calling context ids
+        for OTF2? Perhaps there should also be a way to map entry rows to
```
Comment (hsirkar) [inline pipit/trace.py]: This is resolved by `match_rows()` function from #44 right?

**Comment by bhatele** [comment]
Lets make the CCT construction optional (dependent on user input)

**Review by hsirkar — APPROVED** [review]

**Review by bhatele — APPROVED** [review]

**Pull request merged**